feat: 식당 정보 가져오는 스케쥴링 기능 추가

HyunsooZo opened the pull request:
### 작업 내용
- 식당 정보 가져오는 스케쥴링 기능 추가
- 프로젝트 내 중복파일 삭제

### 변경 사항(추가시엔 추가사항)
- ***[요건 B-데이터 수집]*** 
       총 5개 외부 api를 호출합니다. (경기도 일반음식점 _ 패스트푸드,중식,양식,뷔페,일식) 
- ***[요건 B-데이터 전처리]*** 
       모든 값을 그대로 저장하되 `null`값은 데이터 타입에 따라 `데이터없음` , `0` , `0.0` 으로 전처리                                   
-  ***[요건 B-데이터 저장]*** 
    유일키는 식당이름+지번주소 에 공백을 제거하여 사용                
-  ***[기타 커스텀 조건]*** 
   `폐업`상태 식당의 경우 저장하지 않음.
     매일 `23:59` 스케줄링 동작
     저장시점에 저장 식당 종류, 시간을 로깅
     이미 저장된 식당의 경우 업데이트 진행

### 특이 사항
- 테스트코드 작성에 적합하지 않은 기능이라 스케쥴링 테스트 및 DB 적재 확인만 진행

### 테스트
- [ ] 테스트 코드
- [x] api 테스트

### 관련 이슈(옵셔널)
- 없음

HyunsooZo committed:
chore: JSON 파싱을 위한 jackson 라이브러리 의존성 추가

HyunsooZo committed:
remove: 프로젝트 내 중복파일 삭제

HyunsooZo committed:
chore: 스케줄링 및 BaseEntity 적용을 위한 어노테이션 추가

HyunsooZo committed:
feat: 식당 정보 가져오는 스케쥴링 기능 추가

총 5개 외부 api를 호출합니다. (패스트푸드,중식,양식,뷔페,일식)
- 모든 값을 그대로 저장하되 null값은 데이터 타입에 따라 "데이터없음" , 0 , 0.0 등 으로 전처리
- 유일키는 식당이름+지번주소 에 공백을 제거하여 사용
- "폐업" 식당의 경우 저장하지 않음.
- 매일 23:59 스케줄링 동작
- 저장시점에 저장 식당 종류, 시간을 로깅

madst0614 reviewed (approved):
주석도 잘 달려있어서 이해하기 편하고 전체적으로 깔끔하게 잘 적어주신 것 같습니다!

HyunsooZo committed:
feat: 기존 식당 정보 존재 시 업데이트 하는 기능 추가

영업점상태(폐업,영업)
- 폐업시 삭제 영업시 업데이트
Restaurant 필드 추가
- (BaseEntity 업데이트,생성일)
- 평점 , 리뷰 개수

HyunsooZo committed:
style: 가독성을 위한 줄바꿈 및 불필요한 어노테이션 삭제

junkyu92 reviewed (approved):
데이터가 없을 경우의 처리까지 잘 하신 것 같습니다. 👍

12hyeon reviewed (approved):
코드가 깔끔하고, 이해하기 좋았습니다!
